### PR TITLE
feature #24 : added a button that set the map to the current location

### PR DESCRIPTION
**Pull request by marwan011**

### CODE REVIEW

**Comment by marwan011** [comment]
Adding a  current position button that set the map view to the current location.
Updates current location when moving
Shows a current location pin

File: App.tsx [hunk 1]
```diff
@@ -76,21 +107,54 @@ class App extends Component<{}, appState> {
...
   };
 
+  success = (location: any) => {
+    const { position } = this.state;
+    position.setLatitude(location.coords.latitude);
+    position.setLongitude(location.coords.longitude);
+    this.setMapLocation(position);
+  };
+
+  failure = (err: any) => {
```
Comment (Alessandro100) [inline App.tsx]: whats the purpose of this function if it doesn't do anything

File: App.tsx [hunk 2]
```diff
@@ -76,21 +107,54 @@ class App extends Component<{}, appState> {
     });
   };
 
+  success = (location: any) => {
```
Comment (Alessandro100) [inline App.tsx]: why is this called 'success' can you name it something with more meaning

File: App.tsx [hunk 3]
```diff
@@ -76,21 +107,54 @@ class App extends Component<{}, appState> {
...
             startLocation={new OutdoorPOI(new Location(45.458488, -73.639862), 'test-start')}
             endLocation={new OutdoorPOI(new Location(45.50349, -73.572182), 'test-end')}
             transportType={transportMode.transit}
           />
         </MapView>
+        <View style={styles.positionBtn}>
+          <TouchableOpacity onPress={() => this.displayCurrentLocation()}>
+            <Image style={styles.iconSize} source={require('./assets/cp.png')} />
+          </TouchableOpacity>
+        </View>
```
Comment (Alessandro100) [inline App.tsx]: This should be a separate component called 'GetLocationButton' and you would pass the 'setMapLocation' function from App.tsx. Look at my ToggleCampusButton Component for reference. Yours should look something like 

<GetLocationButton setMapLocation={this.setMapLocation} />

The purpose of this is to really make our app into components which helps with building logic

File: App.tsx [hunk 4]
```diff
@@ -76,21 +107,54 @@ class App extends Component<{}, appState> {
...
+    console.log(err);
+  };
+
+  displayCurrentLocation() {
+    const options = {
+      enableHighAccuracy: true,
+      timeOut: 20000,
+      maximumAge: 60 * 60 * 24,
+    };
+    navigator.geolocation.getCurrentPosition(this.success, this.failure, options);
```
Comment (Alessandro100) [inline App.tsx]: instead of having 'this.failure' which doesn't do anything, just replace it with null

**Comment by Alessandro100** [comment]
Nice fixes, component looks good

**Pull request merged**